pulsar/internal/connection.go

```diff
@@ -697,6 +697,16 @@ func (c *connection) deletePendingRequest(requestID uint64) (*request, bool) {
 	return request, ok
 }
 
+func (c *connection) failPendingRequests(errMsg string) bool {
```

Comment:
Is the return value used?

There is some duplicate code around line 370 (https://github.com/apache/pulsar-client-go/blob/master/pulsar/internal/connection.go#L370) that might benefit from this function. Maybe it can take an `err Error` as an argument instead of a string.

Reply:
done